src/main/java/com/hcl/appscan/jenkins/plugin/scanners/StaticAnalyzer.java

```diff
@@ -77,29 +90,84 @@ public String checkScanSpeed(String scanSpeed) {
         		}
         	return null;
     	}
-       
-        public boolean isOpenSourceOnly() {
-            if(!m_scanMethod.equals(CoreConstants.UPLOAD_DIRECT)){
-                return m_openSourceOnly;
-            }
-            return false;
-        }
-        
+
         @DataBoundSetter
         public void setOpenSourceOnly(boolean openSourceOnly) {
             m_openSourceOnly = openSourceOnly;
         }
 
-        public boolean isSourceCodeOnly() {
+        public boolean getOpenSourceOnly() {
+            return m_openSourceOnly;
+        }
+        
+        @DataBoundSetter
+        public void setIncludeSCAGenerateIRX(String includeSCAGenerateIRX) {
+            if(getHasOptions() && includeSCAGenerateIRX.equals("true")) {
```

Comment:
This can be simplified to:
`return getHashOptions() && Boolean.parseBoolean(includeSCAGenerateIRX);`

Reply:
Okay, sure